#24-Singly-linked-lists-and.md

```diff
@@ -0,0 +1,24 @@
+
+### Singly Linked Lists
+Singly linked lists’ nodes are connected only to the next node. Singly linked lists are great when you want to access data in the forward direction only. The insertion and deletion of nodes of nodes is easy. Singly linked list require less memory than doubly linked lists. However, the main problem with singly linked lists is that directly accessing the previous node is impossible. You experience examples of singly linked list when visiting a bank because the teller is typically required to process the customers in the order they arrive in. The problem with singly linked list becomes obvious when told to rewind a song but only the fast forward "button" works. 
+
+### Doubly Linked Lists
+In contrast, doubly linked lists’ nodes are connected to both the previous and next node and this difference is very important. The main advantage of a doubly linked list over a singly linked list is its ability to traverse easily both in a forward and backward directions. This means you now have the ability ot rewind the song to the beginnging. But doubly linked lists can be disadvantageous compared to singly linked lists as well. With regards to programming, extra space is needed to store the extra pointer in a doubly linked list. Additionally, the extra pointer can be burdensome, as all operations such as insertion would require more logic to support the need to point to both the next and previous nodes. 
+
+## Conclusion
+Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. A bank teller's line is a perfect everyday example of its advantage; and a broken rewind button its disadvantage. The advantaged of doubly linked lists can be demonstrated when having a song that you want to fast forward or rewind -- a feature that singly linked list would not possess. Doubly linked lists are also used to undo operations because it involves a user’s history and can be used to track or determine the executions of multiple processes.  Similar to the doubly linked list, singly linked lists can also be used to undo functions but you would never be able to "redo"  or "undo the undo". 
+
+
+### Conclusion
+ can be problematic or beneficial depending on the circumstance and each have their own unique uses. Linked lists are very important to computer science, and many commonly used programming concepts such as queues and stacks owe their existance to linked lists.
```

Comment:
- Beginning of paragraph is missing
- comma after "circumstance"
- commas around "such as queues and stacks"
- do you want the queues and stacks piece links within this article?